### PR TITLE
Light Cigars when interacting with hot stuff

### PR DESCRIPTION
**Pull request by JustinTether**
Subscribed to IsHotEvent in ThrusterSystem, added AfterInteractEvent to SmokingSystem.Cigar which will set SmokableState to Lit if IsHotEvent returns true

This should mean that cigs/Cigars should light when interacting with anything hot, which should fix issue 5429

## About the PR 

**Screenshots**

**Changelog**

:cl: JustinTime
- tweak: You can now light ciggies by interacting with hot things
- tweak: Thrusters are now hot

### CODE REVIEW

**Comment by DogZeroX** [comment]
You should make the changelog more user-friendly. "You can now light the cigar by clicking it on a thruster."

**Comment by JustinTether** [comment]
Yea that makes sense, especially if it's going to be public facing

File: Content.Server/Shuttles/EntitySystems/ThrusterSystem.cs [hunk 1]
```diff
@@ -95,6 +96,11 @@ public override void Shutdown()
             _mapManager.TileChanged -= OnTileChange;
         }
 
+        private void OnIsHotEvent(EntityUid uid, ThrusterComponent component, IsHotEvent args)
+        {
+            args.IsHot = component.Enabled;
+        }
```
Comment (metalgearsloth) [inline Content.Server/Shuttles/EntitySystems/ThrusterSystem.cs:102]: Gyros shouldn't be hot IMO, for now just check if it's an angular thruster.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Nutrition/EntitySystems/SmokingSystem.Cigar.cs [hunk 2]
```diff
@@ -51,6 +52,21 @@ private void OnCigarInteractUsingEvent(EntityUid uid, CigarComponent component,
...
         }
 
+        public void OnCigarAfterInteract(EntityUid uid, CigarComponent ownerCigarComponent, AfterInteractEvent args)
+        {
+            var targetEntity = args.Target;
+            if (targetEntity == null)
+                return;
+
+            var isHotEvent = new IsHotEvent();
+            RaiseLocalEvent(targetEntity.Uid, isHotEvent);
```
Comment (metalgearsloth) [inline Content.Server/Nutrition/EntitySystems/SmokingSystem.Cigar.cs:62]: Your branch is very old and needs updating hence the error.

**Comment by JustinTether** [comment]
I think 99cb6a0 should address the issues brought up by metalgearsloth

**Review by metalgearsloth — APPROVED** [review]
I made some changes:
- Check IsOn not Enabled (as that is the toggle not whether the thruster is lit)
- Check if smokable already lit up front to avoid the eventbus event.

**Pull request merged**